### PR TITLE
Upgrade ArgoCD to v2.3.0-rc5

### PR DESCRIPTION
**Pull request by terrytangyuan**
Remaining diff:
```
17235a17236
> data: null
17335a17337
>   annotations: null
17436a17439
>   annotations: null
18347,18348d18349
<         - name: ARGOCD_CONTROLLER_REPLICAS
<           value: "1"
```

TODOs for this PR:

- [x] Remove applicationset and notifications charts
- [x] Check if we have parameterized enough, especially for applicationset and notifications.
- [x] Address additional comments

Things that are out-of-scope in this PR:
- Fix CI checks (we cannot fix it since it failed because HA requirement in the manifests)
- Include image updater as an umbrella chart (a separate PR since the current image updater chart needs to be revamped as well)

TODOs when the official release for v2.3.0 is ready:
* The env var `ARGOCD_CONTROLLER_REPLICAS` will be fixed in upstream.
* There were empty fields and missing fields in ApplicationSet CRD in upstream. It's fixed in upstream now. 
* This commit https://github.com/akuity/helm-charts/commit/0d3cdf2141c868c0975a8fdf43e66e6e5cabd4da (besides the added missing argocd-applicationset-controller service account) will need to be reverted if we want to upgrade to future versions of `v2.3.0` since the original manifest was from master branch.

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -115,6 +115,13 @@ repoServer:
     #   cpu: 10m
     #   memory: 64Mi
 
+# -- ApplicationSet Controller
+applicationsetController:
+  image:
+    repository: quay.io/argoproj/argocd-applicationset # defaults to global.image.repository
```
Comment (terrytangyuan) [inline charts/argo-cd/values.yaml:121]: This needs to be on akuity Quay repo https://github.com/akuityio/akuity-enterprise/issues/18

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ global:
     # -- If defined, a repository applied to all ArgoCD deployments
     repository: quay.io/akuity/argocd
     # -- If defined, a tag applied to all ArgoCD deployments
-    tag: v2.2.4-ak.0
+    tag: v2.3.0-rc5-ak.0
```
Comment (terrytangyuan) [inline charts/argo-cd/values.yaml:9]: Not released yet.

File: hack/compare-cd.sh [hunk 3]
```diff
@@ -29,4 +32,4 @@ upstream_out=$(
     grep -v "This is an auto-generated file"
 )
 
-diff <(echo "$helm_out") <(echo "$upstream_out")
+diff <(echo "$helm_out") <(echo "$upstream_out") | subl -
```
Comment (terrytangyuan) [inline hack/compare-cd.sh:35]: Remove debugging

File: hack/compare-cd.sh [hunk 4]
```diff
@@ -12,8 +12,11 @@ helm template \
     --include-crds \
     --set global.image.repository=quay.io/argoproj/argocd \
     --set global.image.tag=${upstream_version} \
+    --set config.argocd=null \
```
Comment (terrytangyuan) [inline hack/compare-cd.sh:15]: Need to figure out a generic solution for `configmap.data`

File: charts/argo-cd/values.yaml (hunk 1 above)
Comment (rbreeze) [reply charts/argo-cd/values.yaml:128]: Should we change this in a separate PR?

Comment (terrytangyuan) [reply charts/argo-cd/values.yaml:128]: I think so. I left this as a separate issue for discussion https://github.com/akuityio/akuity-enterprise/issues/18.

File: charts/argo-cd/Chart.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
-version: 1.0.5
-appVersion: 2.2.4
+version: 1.0.6
```
Comment (jessesuen) [inline charts/argo-cd/Chart.yaml:2]: We are jumping versions, so this should be a 1.1.0 release

File: charts/argo-cd/values.yaml [hunk 6]
```diff
@@ -115,6 +115,20 @@ repoServer:
...
 
+# -- Notifications Controller
+notificationsController:
+  # -- Whether to enable Notifications Controller
+  enabled: true
+
+# -- ApplicationSet Controller
+applicationsetController:
+  # -- Whether to enable ApplicationSet Controller
+  enabled: true
```
Comment (jessesuen) [inline charts/argo-cd/values.yaml:126]: I think the default should actually be disabled

Comment (terrytangyuan) [reply charts/argo-cd/values.yaml:126]: Then the defaults are different from upstream and we have to explicitly set this to true when running our diff comparison script.

File: charts/argo-cd/Chart.yaml (hunk 5 above)
Comment (terrytangyuan) [reply charts/argo-cd/Chart.yaml]: Good catch. Updated.

**Review by rbreeze — APPROVED** [review]
LGTM

**Review by jessesuen — APPROVED** [review]

**Pull request merged**